rs/nns/governance/api/src/pb.rs

```diff
@@ -95,3 +113,154 @@ impl XdrConversionRate {
         }
     }
 }
+
+// The following methods are conceptually methods for the API type of the neuron.
+impl Neuron {
+    pub fn state(&self, now_seconds: u64) -> NeuronState {
+        if self.spawn_at_timestamp_seconds.is_some() {
+            return NeuronState::Spawning;
+        }
+        match self.dissolve_state {
+            Some(DissolveState::DissolveDelaySeconds(dissolve_delay_seconds)) => {
+                if dissolve_delay_seconds > 0 {
+                    NeuronState::NotDissolving
+                } else {
+                    NeuronState::Dissolved
+                }
+            }
+            Some(DissolveState::WhenDissolvedTimestampSeconds(
+                when_dissolved_timestamp_seconds,
+            )) => {
+                if when_dissolved_timestamp_seconds > now_seconds {
+                    NeuronState::Dissolving
+                } else {
+                    NeuronState::Dissolved
+                }
+            }
+            None => NeuronState::Dissolved,
+        }
+    }
+
+    pub fn dissolve_delay_seconds(&self, now_seconds: u64) -> u64 {
+        match self.dissolve_state {
+            Some(DissolveState::DissolveDelaySeconds(dissolve_delay_seconds)) => {
+                dissolve_delay_seconds
+            }
+            Some(DissolveState::WhenDissolvedTimestampSeconds(
+                when_dissolved_timestamp_seconds,
+            )) => when_dissolved_timestamp_seconds.saturating_sub(now_seconds),
+            None => 0,
+        }
+    }
+
+    pub fn stake_e8s(&self) -> u64 {
+        neuron_stake_e8s(
+            self.cached_neuron_stake_e8s,
+            self.neuron_fees_e8s,
+            self.staked_maturity_e8s_equivalent,
+        )
+    }
+}
+
+fn neuron_stake_e8s(
+    cached_neuron_stake_e8s: u64,
+    neuron_fees_e8s: u64,
+    staked_maturity_e8s_equivalent: Option<u64>,
+) -> u64 {
+    cached_neuron_stake_e8s
+        .saturating_sub(neuron_fees_e8s)
+        .saturating_add(staked_maturity_e8s_equivalent.unwrap_or(0))
+}
```

Comment:
This function can be inlined. I think it was created for the API type and domain type to share the function (and the domain type should be able to inline too)